lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,243 @@
+            } else if val.is_i64() {
+                number = val.as_i64().unwrap();
+            }
+            Ok(number)
+        } else if val.is_string() {
+            match val.as_str().unwrap().parse::<i64>() {
+                Ok(number) => Ok(number),
+                Err(_) => match val.as_str().unwrap().parse::<f64>() {
+                    Ok(number) => Ok(f64::round(number) as i64),
+                    Err(_) => Err(format!(
+                        "Event field {} does not match GELF spec version {}: must be a number",
+                        VERSION, GELF_VERSION
+                    )
+                    .into()),
+                },
+            }
+        } else {
+            Err(format!(
+                "Event field {} does not match GELF spec version {}: must be a number",
+                VERSION, GELF_VERSION
+            )
+            .into())
+        }
+    }
+
+    /// Builds a LogEvent from the parsed GelfMessage.
+    /// The logic does not follow strictly the documented GELF standard, it more closely
+    /// follows the behavior of graylog itself, which is more relaxed.
+    fn message_to_event(&self, parsed: &GelfMessage) -> vector_core::Result<Event> {
+        if parsed.message.is_none() && parsed.short_message.is_none() {
+            return Err("Event must contain the field 'short_message'".into());
+        }
+
+        let message = if let Some(msg) = &parsed.short_message {
+            msg
+        } else {
+            parsed.message.as_ref().unwrap()
+        };
+        let mut event = Event::from(message.to_string());
+        let log = event.as_mut_log();
+
+        let timestamp_key = log_schema().timestamp_key();
+
+        // GELF spec defines the version as 1.1 which has not changed since 2013
+        // but graylog server does not reject any event which does not specify a version
+
+        if let Some(host) = &parsed.host {
+            log.insert(HOST, host.to_string());
+        }
+        // TODO graylog sets field 'host' to IP address if not specified. I'm not seeing a clear way to get the IP...
+
+        // FACILITY, FILE, FULL_MESSAGE can be any type and are optional
+
+        if let Some(add) = &parsed.additional_fields {
+            for (key, val) in add.iter() {
+                // Additional field names must be characters dashes or dots.
+                // Drop fields names that contain offending characters.
+                if !self.regex.is_match(key) {
+                    continue;
+                }
+
+                // if the timestamp is not numeric, we set it to current UTC later
+                if key == TIMESTAMP && val.is_number() {
+                    log.insert(timestamp_key, self.parse_timestamp(val));
+                }
+                // level and line are both numbers
+                else if key == LEVEL || key == LINE {
+                    log.insert(path!(key), self.parse_number(val)?);
+                } else {
```

Comment:
Ah good call! I had these originally in the GelfMessage (before relaxing the parsing strictness) , and removed because of not wanting to restrict the type- didn't realize could do the serde Value in the decoded messge 👍